scenario/go/scenario_test.go

```diff
@@ -38,10 +38,12 @@ const (
 const defaultListenerName = ""
 
 var (
-	gateways                 map[string]*GatewayConnection
-	currentGateway           *GatewayConnection
-	transaction              *Transaction
-	lastCommittedBlockNumber uint64
+	gateways                   map[string]*GatewayConnection
+	currentGateway             *GatewayConnection
+	transaction                *Transaction
+	lastCommittedBlockNumber   uint64
+	checkpointer               *client.InMemoryCheckpointer
+	lastChaincodeEventReceived *client.ChaincodeEvent
```

Comment:
This shouldn't be needed as checkpointing should happen as events are read rather than as a discreet step after reading